### DISCUSSION TITLE
Creating a gist with the API returns 500

### DISCUSSION THREAD

**Discussion by seinecle** [community/community#40628] [Apps, API and Webhooks]
# Issue
since end of November 2022 a call to the api endpoint for the creation of a gist returns a 500 code, when it worked perfectly before.

# Context
Gephi is an open source software extensible with plugins.
One of such plugins exports a file created with Gephi to a public gist.
Until a few days ago, the call to create a gist was successful. Then around Nov 28, users started reporting that the plugin failed with an code 500 error.

# Attempts to fix
As the error came right at the time when Github introduced versioned APIs, I thought this could be related, even though the change is supposed not to be breaking. I [added the header](https://github.com/gephi/gephi-plugins/commit/4592e8d3f64b81757bc64107019a6e761c4f06cc#diff-47c8528e401b7aa7403c30053e50d1b9463c1f7b63889b457538dc74c813b5f8R196) that accounts for this versioning, with no result.

# To investigate
- the [documentation](https://docs.github.com/en/rest/gists/gists?apiVersion=2022-11-28#create-a-gist) for the API endpoint for gist creation.
- the [code making the call](https://github.com/gephi/gephi-plugins/blob/4592e8d3f64b81757bc64107019a6e761c4f06cc/modules/WebPublishPlugin/src/main/java/net/clementlevallois/web/publish/plugin/github/PublishRunnable.java#L192) to the API endpoint

Thank you for your help, many users depend on this feature to be working again.

**Comment by ossanna16** [comment]
Hi there @seinecle and welcome to our community! Thank you for asking a great question 🙂
While you are waiting for a reply to your question, here are a few things you can do:

To get started, introduce yourself in our [official introduction thread](https://github.com/orgs/community/discussions/23204)
To connect with other new users, check out this Discussion to [Meet Your GitHub Community Team](https://github.com/orgs/community/discussions/28776), further your GitHub knowledge by working through our [GitHub Skill 🆙 exercises](https://github.com/orgs/community/discussions/32121), and participate in our [weekly Fun Friday chats](https://github.com/orgs/community/discussions/30912)

**Reply by seinecle** [reply]
Hi @ossanna16 !

It has been a month and our community of users can't use the nice service we developed for them, because of this error we can't solve. Not sure but the 500 error code suggests there is something to investigate on Github's side. We don't have any error message on our side that would tell us more.

To recall: basically, we make a call to the Github API to publish a gist. It worked fine for hundreds of times for a month, and it stopped working abruptly near end of November, returning a 500 error code. What could have happened? Thanks!

**Reply by ossanna16** [reply]
@ghostinhershell Are you able to direct @seinecle to the right person to help with this request?

**Reply by seinecle** [reply]
@ossanna16 @ghostinhershell any progress by any chance?

**Comment by mbastian** [comment]
Hi folks, I'm having exactly the same issue. Any luck resolving this? Thanks!

**Reply by mbastian** [reply]
While debugging it seems that the POST call to `https://github.com/login/oauth/access_token` returns an empty `scope` which doesn't seem normal given the [documentation](https://docs.github.com/en/developers/apps/building-oauth-apps/authorizing-oauth-apps#response-2). I would expect the `scope` to contain `gist` as this is the scope that was requested.

Without the proper scope it may explain why the API then fails with a 500.

**Comment by mbastian** [comment]
For the record, I solved this issue by using an OAuth App instead of a GitHub app. Even though the authentification workflow is the same it works fine for an OAuth App but has this 500 error issue with a GitHub app.
